Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportApiHandler.cs

```diff
@@ -78,243 +81,273 @@ public AccountReportRunSet MapDataToAccountReportObject(LiteDbRunSet runSet)
-            try
+            if (restClient != null)
             {
-                IRestResponse response = await restClient.ExecuteTaskAsync(restRequest);
-                if (response.IsSuccessful)
+                RestRequest restRequest = (RestRequest)new RestRequest(SEND_ACTION_EXECUTION_DATA, isUpdate ? Method.PUT : Method.POST) { RequestFormat = RestSharp.DataFormat.Json }.AddJsonBody(accountReportAction);
+                // restRequest.Parameters.Add(new Parameter("IsUpdate", isUpdate, ParameterType.HttpHeader));
+                string message = string.Format("execution data to Central DB for the Action:'{0}' (Execution Id:'{1}', Parent Activity Id:'{2}')", accountReportAction.Name, accountReportAction.Id, accountReportAction.AccountReportDbActivityId);
+                try
                 {
-                    Reporter.ToLog(eLogLevel.DEBUG, "Successfully sent " + message);
+                    IRestResponse response = await restClient.ExecuteTaskAsync(restRequest);
+                    if (response.IsSuccessful)
+                    {
+                        Reporter.ToLog(eLogLevel.DEBUG, "Successfully sent " + message);
+                    }
+                    else
+                    {
+                        Reporter.ToLog(eLogLevel.ERROR, "Failed to send " + message + "Response: " + response.Content);
+                    }
                 }
-                else
+                catch (Exception ex)
                 {
-                    Reporter.ToLog(eLogLevel.ERROR, "Failed to send " + message + "Response: " + response.Content);
+                    Reporter.ToLog(eLogLevel.ERROR, "Exception when sending " + message, ex);
                 }
             }
-            catch (Exception ex)
-            {
-                Reporter.ToLog(eLogLevel.ERROR, "Exception when sending " + message, ex);
-            }          
         }
 
         public bool ExecutionIdValidation(Guid executionId)
         {
-            RestRequest restRequest = (RestRequest)new RestRequest(EXECUTION_ID_VALIDATION + executionId, Method.GET);           
-            string message = string.Format("execution id : {0}", executionId);
-            try
+            if (restClient != null)
             {
-                IRestResponse response = restClient.Execute(restRequest);
-                if (response.IsSuccessful)
+                RestRequest restRequest = (RestRequest)new RestRequest(EXECUTION_ID_VALIDATION + executionId, Method.GET);
+                string message = string.Format("execution id : {0}", executionId);
+                try
                 {
-                    Reporter.ToLog(eLogLevel.DEBUG, "Successfully validated execution id " + message);
-                    return Convert.ToBoolean(response.Content);
+                    IRestResponse response = restClient.Execute(restRequest);
```

Comment:
Please make Execute as ExecuteAsync

Reply:
This was not async from before only, I only added null check. We cannot make it async as the caller expects synchronous response.